recipes/djinni-support-lib/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+    topics = ("java", "Objective-C", "Android", "iOS")
+    license = "Apache-2.0"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+                "fPIC": [True, False],
+                "target": ["JNI", "OBJC", "Auto"],
+                "system_java": [True, False]
+               }
+    default_options = {"shared": False,
+                        "fPIC": True ,
+                        "target": "Auto",
+                        "system_java": False
+                       }
+    exports_sources = ["patches/**", "CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def objc_support(self):
+        if self.options.target == "Auto":
+            return self.settings.os in ["iOS", "Macos"]
+        else:
+            return self.options.target == "OBJC"
+
+    @property
+    def jni_support(self):
+        if self.options.target == "Auto":
+            return self.settings.os != "iOS"
+        else:
+            return self.options.target == "JNI"
+
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
+            del self.options.fPIC
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        if not self.options.system_java:
+            self.build_requires("zulu-openjdk/11.0.8@")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(f"djinni-support-lib-{self.version}", self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if not self.options.shared:
+            self._cmake.definitions["DJINNI_STATIC_LIB"] = True
+        self._cmake.definitions["DJINNI_WITH_OBJC"] = self.objc_support
+        self._cmake.definitions["DJINNI_WITH_JNI"] = self.jni_support
+        if self.jni_support:
+            self._cmake.definitions["JAVA_AWT_LIBRARY"] = "NotNeeded"
+            self._cmake.definitions["JAVA_AWT_INCLUDE_PATH"] = "NotNeeded"
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)        
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        #self.cpp_info.libdirs = ["lib"] # thats default anyway
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = ["include",]
```

Comment:
Not required to set it to the default

```suggestion
        self.cpp_info.libs = tools.collect_libs(self)
```